Source/DafnyCore/AST/BuiltIns.cs

```diff
@@ -233,7 +233,15 @@ public class BuiltIns {
     argumentGhostness ??= new bool[dims].Select(_ => false).ToList();
     if (!tupleTypeDecls.TryGetValue(argumentGhostness, out var tt)) {
       Contract.Assume(allowCreationOfNewType);  // the parser should ensure that all needed tuple types exist by the time of resolution
-      tt = new TupleTypeDecl(argumentGhostness, SystemModule, DontCompile());
+
+      var nonGhostDims = argumentGhostness.Count(x => !x);
+      TupleTypeDecl nonGhostTupleTypeDecl = null;
+      if (nonGhostDims != dims && (nonGhostDims != 1 || !DafnyOptions.O.Compiler.OptimizesInvisibleDatatypeWrappers)) {
```

Comment:
It will just generate a `Tuple1` class as well. That seems alright. I'll change the test to just `nonGhostDims != dims`.

Reply:
The suggested change worked. However, in the end, I decided to revert it to the code above. That is, the test does include `nonGhostDims != 1 || !DafnyOptions.O.Compiler.OptimizesInvisibleDatatypeWrappers` again, so that we will _not_ generate a `Tuple1` class. It would seem odd to generate a class that we know is never going to be used.